src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -623,27 +625,30 @@ private boolean checkForChangedProducts(Product incomingProduct, Pool existingPo
             productChanged = incomingProduct != null;
         }
 
-        if (productChanged) {
+        if (productChanged || force) {
             existingPool.setProduct(incomingProduct);
+            return true;
         }
 
-        return productChanged;
+        return false;
     }
 
-    private boolean checkForDateChange(Date start, Date end, Pool existingPool) {
+    private boolean checkForDateChange(Date start, Date end, Pool existingPool, boolean force) {
         boolean datesChanged = (!start.equals(existingPool.getStartDate())) ||
             (!end.equals(existingPool.getEndDate()));
 
-        if (datesChanged) {
+        if (datesChanged || force) {
```

Comment:
Maybe we should check force and short circuit so we can skip the datesChanged calculation.

Reply:
Fixed